sdk/communication/Azure.Communication.CallAutomation/src/Models/HoldResult.cs

```diff
@@ -0,0 +1,85 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Threading.Tasks;
+using System;
+using System.Threading;
+
+namespace Azure.Communication.CallAutomation
+{
+    /// <summary>The result from playing audio.</summary>
+    public class HoldResult
+    {
+        private CallAutomationEventProcessor _evHandler;
+        private string _callConnectionId;
+        private string _operationContext;
+
+        internal HoldResult()
+        {
+        }
+
+        internal void SetEventProcessor(CallAutomationEventProcessor evHandler, string callConnectionId, string operationContext)
+        {
+            _evHandler = evHandler;
+            _callConnectionId = callConnectionId;
+            _operationContext = operationContext;
+        }
+
+        /// <summary>
+        /// This is blocking call. Wait for <see cref="HoldEventResult"/> using <see cref="CallAutomationEventProcessor"/>.
+        /// </summary>
+        /// <param name="cancellationToken">Cancellation Token can be used to set timeout or cancel this WaitForEventProcessor.</param>
+        /// <returns>Returns <see cref="HoldEventResult"/> which contains either <see cref="PlayCompleted"/> event or <see cref="PlayFailed"/> event.</returns>
+        public HoldEventResult WaitForEventProcessor(CancellationToken cancellationToken = default)
```

Comment:
For Hold Event, there is no success case? 
how would this continue without success event coming bacK?

Reply:
Currently we dont have any event introduced for the hold success, we can check on the participant to see whether he is on hold or not